### ISSUE TITLE
inconsistent tilde (~) handling

### ISSUE THREAD

**Issue by rbrewer123** [scopatz/xonsh#260]
On Windows, handling of tilde (``~``) expansion is inconsistent.  xonsh seems to expand tilde correctly (to the user's home directory) for its builtin functions such as ``cd ~``.

Windows commands perform no tilde expansion, so for example, doing ``move foobar.txt ~`` results in renaming the file ``foobar.txt`` to a file named ``~`` in the current directory.

Unix commands ported to Windows in the msysGit suite (e.g. ``cp``, ``mv``) perform their own tilde expansion.  Unfortunately, they expand tilde to the location where msysGit is installed.

To summarize: on windows there are currently 3 different behaviors of tilde expansion.  It seems like this could be cleaned up if on Windows xonsh always performed tilde expansion of all arguments in subprocess mode.  Does this seem like a valid way to proceed?

**Comment by wrywerytwreywery** [comment]
This seems to happen on GNU/Linux as well, so I don't think it is specific to Windows.  I think your suggestion is the right way.  Right now, the expansion is happening at a weird time, so that single characters never get expanded (`cd` only works because it also handles the `~` character itself (it never gets expanded to the user's home)).

That expansion is currently happening [here](https://github.com/scopatz/xonsh/blob/master/xonsh/parser.py#L2299-2309), but it should probably be moved someplace else, after we have the full `subproc_atom` (maybe in `p_subproc_atom`, [here](https://github.com/scopatz/xonsh/blob/master/xonsh/parser.py#L2262-2263)?).  I'm happy to make the change myself, or to help if you want to tackle it.

**Comment by rbrewer123** [comment]
I haven't looked into the parser at all yet so it's probably best if you take this one.

**Comment by scopatz** [comment]
@rbrewer123 hopefully @wrywerytwreywery's fix worked

**Comment by rbrewer123** [comment]
Finally got a chance to test this and it is working well.  Thanks @wrywerytwreywery!
